src/xdr/Stellar-transaction.x

```diff
@@ -308,7 +308,7 @@ case MEMO_NONE:
 case MEMO_TEXT:
     string text<28>;
 case MEMO_ID:
-    uint64 id;
+    int64 id;
```

Comment:
Why is this safe? The memo is user-specified, so this could have been `UINT64_MAX`.

Reply:
Reverted this (i think you are right, though I saw some sdks using signed ints)